1_download_dialogs_data.py

```diff
@@ -262,4 +341,7 @@ async def download_dialog(client, id, MSG_LIMIT, config):
         print(f"Loading dialog #{id}")
 
         with client:
-            client.loop.run_until_complete(download_dialog(client, id, MSG_LIMIT, config))
+            if(CONTINUATION):
```

Comment:
Logic in this block doesn't work
download_dialog - expects 4 args, you give 5

did you try to run your code?

please double check it works,
remove "CONTINUATION" variable

logic should be either do it all in one or go by batches

default logic should be to go by batches with restore from the last step

Reply:
I believe I fixed it in a better way than you suggested. Please take a look on my update and leave feedback